hazelcast/src/main/java/com/hazelcast/config/SchemaViolationConfigurationException.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
```

Comment:
I think this can stay here. The user may want to catch the exception and in that case it is better if they don't have to import it from an internal package. WDYT?

Reply:
Hm, could be. Can you mention a use case where catching this concrete exception instead of its parent `InvalidConfigurationException` would be desired? In general, I prefer being conservative on introducing a public API. My way of thinking: make everything internal API unless we are aware of a use case that certainly helps the user. If we're unsure, it's better to be internal. If someone comes up with a meaningful reason to make an internal API public, we can always do that, while we can't do the same the other way around. So I see this as an `InvalidConfigurationException`, with specific content that's going to be logged anyways. I'm not sure a user is willing to ask the concrete location of the violation programmatically. I might be wrong on this though :smile: